### PR TITLE
JWT enabled decorator

### PR DESCRIPTION
**Pull request by andrii-z4i**
Basic idea is to be able to pass to the controller jwt related information (iss, aud, sub) after passing successfully jwt validation by express-jwt module.

### CODE REVIEW

File: src/decorators/jwt.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export function JWT(): any {
+  return () => { return; };
+}
```
Comment (lukeautry) [inline src/decorators/jwt.ts]: There should be a line ending at the end of the file.

File: src/interfaces/jwtHolder.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export class JwtHolder {
```
Comment (lukeautry) [inline src/interfaces/jwtHolder.ts]: Could this be an interface? Prefer interfaces for pure data structures.

File: src/interfaces/jwtHolder.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+    public sub: string;
+    public iss: string;
+    public aud: string;
+}
```
Comment (lukeautry) [inline src/interfaces/jwtHolder.ts]: EOD line ending

File: src/metadataGeneration/controllerGenerator.ts [hunk 4]
```diff
@@ -4,9 +4,11 @@ import { MethodGenerator } from './methodGenerator';
 
 export class ControllerGenerator {
   private readonly pathValue: string | undefined;
+  private readonly jwtEnabled: boolean = false;
```
Comment (lukeautry) [inline src/metadataGeneration/controllerGenerator.ts]: Remove redundant type annotation, i.e. just

```
private readonly jwtEnabled = false;
```

**Comment by andrii-z4i** [comment]
Thank you @lukeautry for you comments. I addressed them and ensured that tests are passing + added one especially for functionality I'm adding.

**Comment by andrii-z4i** [comment]
I added generation of swagger json security part if we have jwt token decorator for controller.

**Comment by lukeautry** [comment]
@andrii-z4i Thanks again for the work. This has been merged and rolled into 1.0.1.

**Pull request merged**